lnwallet/channel.go

```diff
@@ -8689,18 +8692,28 @@ func (lc *LightningChannel) MaxFeeRate(maxAllocation float64) chainfee.SatPerKWe
 	// exactly why it was introduced to react for sharp fee changes.
 	availableBalance, weight := lc.availableBalance(AdditionalHtlc)
 
-	oldFee := lc.localCommitChain.tip().feePerKw.FeeForWeight(weight)
-
+	currentFee := lc.localCommitChain.tip().feePerKw.FeeForWeight(weight)
 	// baseBalance is the maximum amount available for us to spend on fees.
-	baseBalance := availableBalance.ToSatoshis() + oldFee
-
+	baseBalance := availableBalance.ToSatoshis() + currentFee
+
+	// In case our local channel balance is drained, we make sure we do not
+	// decrease the fee rate below the current fee rate. This could lead to
+	// a scenario where we lower the commitment fee rate as low as the fee
+	// floor although current fee rates are way higher. The maximum fee
+	// we allow should not be smaller then the current fee. The decrease
+	// in fee rate should happen when the mempool reports lower fee levels
+	// rather than us decreasing in local balance. The max fee rate is
+	// always floored by the current fee rate of the channel.
 	maxFee := float64(baseBalance) * maxAllocation
+	feeAllocation := maxAllocation
+	if float64(currentFee) > maxFee {
+		feeAllocation = float64(currentFee) / float64(baseBalance)
+		maxFee = float64(currentFee)
+	}
```

Comment:
I'd prefer to take a more functional approach using [Max](https://pkg.go.dev/math#Max). From there it follows that the feeAllocation is always `maxFee / baseBalance`. This avoids having to track the changing semantics through this if block.

```go
idealMaxFee := float64(baseBalance) * maxAllocation
maxFee := math.Max(float64(currentFee), idealMaxFee)
maxFeeAllocation := maxFee / float64(baseBalance)
maxFeeRate := ...
return maxFeeRate, maxFeeAllocation
```

With this style, each name has a *singular semantic interpretation*.